app/web/src/components/Connection.vue

```diff
@@ -6,28 +6,47 @@
     >
       ID = {{ connection.id }}
     </div>
-    <SocketCard
-      :socket="connection.fromSocket"
-      outputSocket
+    <ConnectionEndCard
+      v-if="isSocketConnection(connection)"
       :changeStatus="connection.changeStatus"
+      :componentName="componentNames.from"
+      :subjectLabel="connection.fromSocket.def.label"
+      type="input-socket"
+    />
+    <ConnectionEndCard
+      v-else-if="isSubscriptionConnection(connection)"
```

Comment:
You should be able to do v-else here (typescript will use inference). That will help if we add more edge types in the future, because this will stop compiling instead of having a dangling else :)

Reply:
(Same with other similar places)